### PR TITLE
You can't break the quantum link on an inverter while it's still recharging, and attempting to do so will have unforeseen consequences

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

You can't break the quantum link on an inverter while it's still recharging, and attempting to do so will have unforeseen consequences.

## Why It's Good For The Game

Effortless no recourse getaway tools are lame. There should be some risk. Throw the other end of the link somewhere dangerous or something and be creative.

## Changelog
:cl:
balance: You can't break the quantum link on an inverter while it's still recharging, and attempting to do so will have unforeseen consequences.
balance: Experimental teleporter technology has its' dangers, after all.
/:cl:

### CODE REVIEW

**Comment by Wallemations** [comment]
I think this should be chance-based like the other teleporting mishaps, but otherwise I see no issue with it personally. 
Remind me since it's been a bit since I've used them, do they show you visually/in exam-text if they're recharging?

**Comment by Serxule** [comment]
can you describe what this actually does? because i have no clue what this does and im assuming its just more salt

**Comment by thgvr** [comment]
> can you describe what this actually does? because i have no clue what this does and im assuming its just more salt

If you use a quantum spin inverter, teleport, and immediately unlink it, the downside of the inverter is completely nullified. The downside in this case is that you leave an inverter behind and it can be used again to bring the other inverter which presumably the person is holding back.

File: code/game/objects/items/devices/swapper.dm [hunk 1]
```diff
@@ -69,6 +70,17 @@
...
+		if(linked_swapper)
+			if(ishuman(user))
+				var/mob/living/carbon/human/homer = user
+				if(!isflyperson(user))
+					to_chat(homer, span_warning("You hear a buzzing in your ears as [src] sparks, attracting a passing fly which gets caught in the quantum flux field!"))
+					homer.set_species(/datum/species/fly)
+					homer.log_message("was turned into a [initial(species_to_transform.name)] through [src].", LOG_GAME)
+				else
+					to_chat(homer, span_warning("[src] sparks, but doesn't attract a fly this time, an error message on the device stating it's still charging and can't safely deconstruct the quantum flux field."))
+		return
```
Comment (JohnFulpWillard) [inline code/game/objects/items/devices/swapper.dm:83]: Do we want it to still not disconnect?

**Review by JohnFulpWillard — CHANGES_REQUESTED** [review]
I'm gonna mark as draft until you get this compiling, however I would like to give two opinions I have:

1- Setting people to fly seems overused, it's done by teleporters and now by genetic infusing, do we have no better alternative?
2- What are we trying to fix by doing this? You want to prevent people from escaping areas easily, but you'll punish other people who wound unsync the inverter so they can't come back instead, a more likely scenario since people can just discard an inverter after use.

I think this PR **can** be good, I just don't agree with this specific direction of taking it.

**Comment by necromanceranne** [comment]
> The only use of this I thought of is other people unlinking the swapper of someone they don't want constantly teleporting into their department, which this PR would punish if they did it at the wrong time. Please have a better descriptive 'Why this is good for the game', I think this PR could be good, but it seems mislead.

That's not how the inverter works. Or, really, if that is happening, that person is getting in some other way (or they're being very devious with cargo chutes). You can't use these effectively as a B&E tool on their own. You're probably confusing them with quantum launchpads.

Inverters only swap locations with their linked paired inverter (including anything that contains those inverters in its contents, like lockers and people). You need to physically relocate the other inverter to swap to that location, and then the one you swapped with ends up where you were. The one you used is on cooldown, the inverter you traded places with isn't. This means that one can then be used to pull the other inverter (and you if you didn't ditch it) back and swap places again.

What goof wants to punish is people using it as a getaway tool by unlinking the inverters from the escapees side. When you unlink the inverters, pursuers can't either give chase if you ditched it, or bring you back if you didn't.

**Comment by JohnFulpWillard** [comment]
I was thinking of inverters because breaking in a department once is easier than multiple times

but also pfft i misread getaway for gateway, I'll update my comment to fix that

**Comment by necromanceranne** [comment]
You're thinking of B&E, which is really a nonfactor for someone who can get into science by their own power. No, the real strength is an effective a global teleport to a safe location, which is much handier for people knee deep in spent bullet casings and corpses.

Not every problem is tide related. A lot more of it is being a dangerous arsehole stuck in a quantum loop to the break room.

(Also, its funnier to just rig the other quantum inverter with a bomb.)

**Comment by Bawhoppen** [comment]
> This change will matter for literally nobody, except new players who might accidentally do this. Any science powergamer will just add slimeperson mutation toxin to their gamer bag, also this is utterly pointless as the cool down is for both inverters is shared and you can just wait for the cool down to be over then unlink. Or you could just ditch it, or space it, or destroy it. Literally pointless and a noob trap

That's a good point, we need to nerf that too then if it's such a freebie gamer loot

**Comment by KittyNoodle** [comment]
> That's a good point, we need to nerf that too then if it's such a freebie gamer loot 

That is not at all what I want or meant. It’s honestly fine where it is right now imo.
It’s a good counter to stun “combat” and is rarely used anyway

**Comment by moocowswag** [comment]
> can you describe what this actually does? because i have no clue what this does and im assuming its just more salt

If you alt click quantum spin inverter while its recharging you get turned into a fly person for free, its good if you want to be a fly person, and not bad if you are actually playing around your quantum spin inverters to use them effectively even if someone gets a hold of the other end of it.

**Comment by Likteer** [comment]
> > can you describe what this actually does? (...)
> 
> _Explains the issue the PR wants to mend_

@thgvr 
I think they wanted to know what the nasty surprise goof wants to add as punishment for people who attempt it. Many others (me) would also want to know.

**Comment by Overuseddart** [comment]
Fly is boring, how about it summons that new paradox clone to kill you if that gets merged, or if not you get distorted in space time and teleport around 3 times like you’re using bluespace crystals in a period over a minute?

**Comment by JohnFulpWillard** [comment]
Open a new if you address concerns and fix CI

**Pull request closed without merging**